newIDE/app/src/VariablesList/index.js

```diff
@@ -133,9 +137,13 @@ export default class VariablesList extends React.Component<Props, State> {
 
     // We don't want to ever manipulate/access to variables that have been deleted (by removeRecursively):
     // that's why it's important to only delete ancestor variables.
-    ancestorOnlyVariables.forEach(({ variable }: VariableAndName) =>
-      variablesContainer.removeRecursively(variable)
-    );
+    ancestorOnlyVariables.forEach(({ name, variable }: VariableAndName) => {
+      variablesContainer.removeRecursively(variable);
+
+      if (this._allVariableNamesSet.has(name)) {
+        this._undefinedVariableNames.add(name);
```

Comment:
Can you explain why we need to mutate this `_undefinedVariableNames` when we remove variables?

Reply:
While computing `_allVariableNamesSet` only once is very useful, I think we could compute `_undefinedVariableNames` in the render method, so it will avoid a bunch of add/delete that are tedious to follow and that we risk to forget after a refactoring.

Also see the other comment: when computing this list of undefined variables, we might have to handle children variables of structures (or arrays)... But let's see that on another comment.